src/index.js

```diff
@@ -116,7 +118,19 @@ function ParseServer({
   const filesController = new FilesController(filesControllerAdapter);
   const pushController = new PushController(pushControllerAdapter);
   const loggerController = new LoggerController(loggerControllerAdapter);
-  
+
+  if (verifyUserEmails) {
+    if (typeof appName !== 'string') {
```

Comment:
This logic doesn't belong into the loading of the entire server and is explicitly designed for MailController here. As we keep adding more things - this method will grow and grow if we use this flow, whilst if we actually do validation on per class creation - it will encapsulate it properly.